docs/tutorials/interaction-tutorials/nft-collection/deploy-contract.md

```diff
@@ -0,0 +1,991 @@
+
+## Amendments made
+
+The first change was to copy and paste the ZRC-6 contract code into this example program and remove some of the comments which caused errors.
+Next, we now need to change the immutable parameters being passed. As usual we always pass ```_scilla_version``` first, followed by our defined parameters.
+
+## Example deployment code
+
+```js
+//  Copyright (C) 2018 Zilliqa
+//
+//  This file is part of Zilliqa-Javascript-Library.
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+const { BN, Long, bytes, units } = require('@zilliqa-js/util');
+const { Zilliqa } = require('@zilliqa-js/zilliqa');
+const {
+  toBech32Address,
+  getAddressFromPrivateKey,
+} = require('@zilliqa-js/crypto');
+
+const zilliqa = new Zilliqa('https://dev-api.zilliqa.com');
+
+// These are set by the core protocol, and may vary per-chain.
+// You can manually pack the bytes according to chain id and msg version.
+// For more information: https://apidocs.zilliqa.com/?shell#getnetworkid
+
+const chainId = 333; // chainId of the developer testnet
+const msgVersion = 1; // current msgVersion
+const VERSION = bytes.pack(chainId, msgVersion);
+
+// Populate the wallet with an account
+const privateKey =
+  '';
+
+zilliqa.wallet.addByPrivateKey(privateKey);
+
+const address = getAddressFromPrivateKey(privateKey);
+console.log(`My account address is: ${address}`);
+console.log(`My account bech32 address is: ${toBech32Address(address)}`);
+
+async function testBlockchain() {
+  try {
+    // Get Balance
+    const balance = await zilliqa.blockchain.getBalance(address);
+    // Get Minimum Gas Price from blockchain
+    const minGasPrice = await zilliqa.blockchain.getMinimumGasPrice();
+
+    // Account balance (See note 1)
+    console.log(`Your account balance is:`);
+    console.log(balance.result);
+    console.log(`Current Minimum Gas Price: ${minGasPrice.result}`);
+    const myGasPrice = units.toQa('2000', units.Units.Li); // Gas Price that will be used by all transactions
+    console.log(`My Gas Price ${myGasPrice.toString()}`);
+    const isGasSufficient = myGasPrice.gte(new BN(minGasPrice.result)); // Checks if your gas price is less than the minimum gas price
+    console.log(`Is the gas price sufficient? ${isGasSufficient}`);
+
+    // Deploy a contract
+    console.log(`Deploying a new contract....`);
+    const code = 
```

Comment:
This can be outdated. Therefore use link and comment to replace the actual codes.